aiohttp/helpers.py

```diff
@@ -402,8 +408,13 @@ def log(self, message, environ, response, transport, time):
         :param float time: Time taken to serve the request.
         """
         try:
-            self.logger.info(self._log_format % self._format_line(
-                [message, environ, response, transport, time]))
+            fmt_info = self._format_line(
+                [message, environ, response, transport, time])
+
+            extra = {name: value for name, value in fmt_info}
+            self.logger.info(self._log_format %
+                             tuple([value for _, value in fmt_info]),
```

Comment:
Aaah, got it.
But if you'll return just `extra` `OrderedDict` from `_format_line` the code may look better.

Also  keys for headers/envvars from dicts like `extra={'%{None}i': '-', '%{SPAM}e': 'EGGS', '%{Content-Length}o': 123, '%{User-Agent}i': 'Mock/1.0'}` looks... Well, they looks non-intuitive.
I would expect something like `input_header[User-Agent]` instead of cryptic `%{User-Agent}i`.

Maybe constructing `_log_format` as format accepting params by name in new-styled `.format` form like `"{status} {remote_address} {input_headers[User-Agent]}"` makes sense?